### DISCUSSION TITLE
Just starting - fist basic SimpleStorage.sol fails..

### DISCUSSION THREAD

**Discussion by frinkemon** [Cyfrin/foundry-full-course-cu#1148] [Q&A]
I've written the very basic code from the course:

// SPDX-License-Identifier: MIT
pragma solidity ^0.8.22; // version prefix version with ^ for newer versions

contract SimpleStorage {
// Types: bool, uint, int, address, bytes, string, bytes, bytes32, int 256, 

    uint256 favouriteNumber;

    function store(uint256 _favouriteNumber) public {

        favouriteNumber = _favouriteNumber;
    }

}

It compiles fine but won't deploy onto Remix VM (London and Merge) and errors with:

creation of SimpleStorage errored: Error occured: invalid opcode.

invalid opcode
	
	The execution might have thrown OR the EVM version used by the selected environment is not compatible with the compiler EVM version.

Debug the transaction to get more information.
[vm]from: 0x5B3...eddC4to: SimpleStorage.(constructor)value: 0 weidata: 0x608...60033logs: 0hash: 0x2da...7963f

However it will deploy on remix Shanghai.

Any idea whats happening here?

**Comment by Niferu** [comment]
Hi @frinkemon 

You just need to change compiler version to match your contract as below:
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/110537615/38ccc07d-3439-4c75-b033-4c9c08d19647)
